fmriprep/workflows/bold/base.py

```diff
@@ -706,89 +677,28 @@ def init_func_preproc_wf(bold_file, ignore, freesurfer,
-        name='ds_summary_report', run_without_submitting=True,
+    ds_report_summary = pe.Node(
+        DerivativesDataSink(suffix='summary'),
+        name='ds_report_summary', run_without_submitting=True,
         mem_gb=DEFAULT_MEMORY_MIN_GB)
 
-    ds_validation_report = pe.Node(
+    ds_report_validation = pe.Node(
         DerivativesDataSink(base_directory=reportlets_dir,
                             suffix='validation'),
-        name='ds_validation_report', run_without_submitting=True,
-        mem_gb=DEFAULT_MEMORY_MIN_GB)
-
-    ds_bold_rois_report = pe.Node(
-        DerivativesDataSink(base_directory=reportlets_dir,
-                            suffix='rois'),
-        name='ds_bold_rois_report', run_without_submitting=True,
-        mem_gb=DEFAULT_MEMORY_MIN_GB)
-
-    def _bold_reg_suffix(fallback, freesurfer):
-        if fallback:
-            return 'coreg' if freesurfer else 'flirt'
-        return 'bbr' if freesurfer else 'flt_bbr'
-
-    ds_bold_reg_report = pe.Node(
-        DerivativesDataSink(base_directory=reportlets_dir),
-        name='ds_bold_reg_report', run_without_submitting=True,
-        mem_gb=DEFAULT_MEMORY_MIN_GB)
-
-    ds_ica_aroma_report = pe.Node(
-        DerivativesDataSink(base_directory=reportlets_dir,
-                            suffix='ica_aroma'),
-        name='ds_ica_aroma_report', run_without_submitting=True,
+        name='ds_report_validation', run_without_submitting=True,
         mem_gb=DEFAULT_MEMORY_MIN_GB)
 
     workflow.connect([
-        (inputnode, ds_bold_reg_report, [
-            ('first_echo' if t2s_coreg else 'source_file', 'source_file')]),
-        (inputnode, ds_summary_report, [('source_file', 'source_file'),
-                                        ('summary_report', 'in_file')]),
-        (inputnode, ds_validation_report, [('source_file', 'source_file'),
-                                           ('validation_report', 'in_file')]),
-        (inputnode, ds_bold_rois_report, [('source_file', 'source_file'),
-                                          ('bold_rois_report', 'in_file')]),
-        (inputnode, ds_bold_reg_report, [
-            ('bold_reg_report', 'in_file'),
-            (('bold_reg_fallback', _bold_reg_suffix, freesurfer), 'suffix')]),
+        (summary, ds_report_summary, [('out_report', 'in_file')]),
+        (bold_reference_wf, ds_report_validation, [
+            ('outputnode.validation_report', 'in_file')]),
     ])
 
-    if use_aroma:
-        workflow.connect([
-            (inputnode, ds_ica_aroma_report, [('source_file', 'source_file'),
-                                              ('ica_aroma_report', 'in_file')]),
-        ])
+    # Fill-in datasinks of reportlets seen so far
+    for node in workflow.list_node_names():
+        if node.split('.')[-1].startswith('ds_report'):
+            workflow.get_node(node).inputs.base_directory = reportlets_dir
+            workflow.get_node(node).inputs.source_file = ref_file
```

Comment:
This seems a little fragile, in the sense that, if somebody forgets to start their datasink node with `ds_report` some time down the way, it may take some time to track down. But if it makes life a lot easier, I guess this is fine.

Reply:
The alternative is passing in the base_directory and the source_file to all depths of workflows (which basically led us to place these datasinks at the top level in first place).

Reply:
Okay, I added this to the code-conventions wiki https://github.com/poldracklab/fmriprep/wiki/Code-conventions

Reply:
Would you feel more confortable now that this constraint is documented?

On the other hand, missing the `ds_report` naming will cause the reportlet to be dismissed and that should be easy to spot on the final report.